src/front/ticketer/lib/pages/page_user_register.dart

```diff
@@ -1,51 +1,237 @@
+      decoration: const InputDecoration(
+          labelText: "First Name", hintText: 'Enter your first name'),
+      validator: (value) {
+        if (value == null || value.isEmpty) {
+          return "Please enter your first name";
+        }
+        return null;
+      },
+    );
+  }
+
+  Widget _lastNameEntryField() {
+    return TextFormField(
+      controller: _lastName,
+      decoration:
+      const InputDecoration(labelText: "Last Name", hintText: 'Enter your last name'),
+      validator: (value) {
+        if (value == null || value.isEmpty) {
+          return "Please enter your last name";
+        }
+        return null;
+      },
+    );
+  }
+
+  Widget _birthDateEntryField() {
+    return TextFormField(
+      controller: _birthDate,
+      decoration:
+      const InputDecoration(labelText: "Birth Date", hintText: 'Choose your birth date'),
+      readOnly: true,
+      onTap: () => _selectDate(context, DateTime.now())
+          .then((date) => {
+            if(date != null)
+              _birthDate.text = '${date.year}-${date.month}-${date.day}'
+          }),
+      validator: (value) {
+        if (value == null || value.isEmpty) {
+          return "Please choose your birth date";
+        }
+        return null;
+      },
+    );
+  }
+
+  Widget _submitButton() {
+    return Container(
+        margin: const EdgeInsets.only(top: 15.0),
+        child: ElevatedButton(
+          onPressed: () => submitUserData(),
+          child: const Text("Submit"),
+        ));
+  }
+
+  Widget _getRegisterContent() {
+    return Form(
+      key: _formKey,
+      child: SingleChildScrollView(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            _titleBanner(),
+            _firstNameEntryField(),
+            _lastNameEntryField(),
+            _birthDateEntryField(),
+            _submitButton(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Text _titleBanner() {
+    return const Text(
+      "Join us as a user",
+      style: TextStyle(
+        fontSize: 22,
+        fontWeight: FontWeight.w600,
+        color: Colors.blueAccent,
+      ),
+    );
+  }
+
+  Future<void> submitUserData() async {
+    if (_formKey.currentState!.validate()) {
+      User user = User(
+          _firstName.text,
+          _lastName.text,
+          _birthDate.text,
+          credentials.email,
+          credentials.password);
+      sendUserRegistrationRequest(user);
 
-  void _incrementCounter() {
-    setState(() {
-      _counter++;
-    });
+      showDialog(
+        context: context,
+        builder: (context) {
+          return AlertDialog(
+            title: const Text("Thank you"),
+            content: const Text("You can sign in into the account now"),
+            actions: [
+              ElevatedButton(
+                onPressed: () => {
+                  Navigator.pushReplacement(
```

Comment:
This shouldn't be pushReplacement but rather pop

Reply:
resolved